src/Objects/MonitorManager.vala

```diff
@@ -174,7 +184,7 @@ public class Display.MonitorManager : GLib.Object {
                 }
             }
         }
-
+        double max_scale = Utils.get_min_compatible_scale (monitors);
```

Comment:
Don't you need to get the *maximum* compatible scale? You will need a new Utils function I think,